src/app/components/album-details/album-details.spec.js

```diff
@@ -0,0 +1,109 @@
+import angular from 'angular';
+import uiRouter from 'angular-ui-router';
+import chai from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import chaiAsPromised from 'chai-as-promised';
+
+chai.use(sinonChai);
+chai.use(chaiAsPromised);
+
+describe('Component: Album Details', function (done) {
+  beforeEach(angular.mock.module('albumDetails'));
+  beforeEach(angular.mock.module(uiRouter));
+ 
+  var element;
+  var scope;
+  beforeEach(angular.mock.inject(function($rootScope, $compile){
+    scope = $rootScope.$new();
+
+    scope.albums = '4MGDnutWZLdTCP3nmT1tgu';
+    element = angular.element('<album-details album="albums"></album-details>');
+    element = $compile(element)(scope);
+    scope.$digest();
+  }));
+ 
+  it('should render the albums details', function() {
+    var component = element[0];
+    chai.expect(component.firstChild.id).to.equal('album-details');
+  });
+ 
+  it('should go back to root', angular.mock.inject(function($location, $state)   {
+    $state.go = sinon.spy();
```

Comment:
Also, in the way that this spy is put, it will overwrite the function, probably is desired here but not on other places.

A better approach will be like this

Like this:
```
let stateGoStub;

beforeEach(angular.mock.inject(($state) => {
   stateGoStub = sinon.stub($state, 'go', angular.noop);
}));

afterEach(() => stateGoStub.restore());

```